criu/include/sched.h

```diff
@@ -0,0 +1,34 @@
+#ifndef __CR_SCHED_H__
+#define __CR_SCHED_H__
+
+#include <linux/types.h>
+
+#ifndef ptr_to_u64
+#define ptr_to_u64(ptr)	((__u64)((uintptr_t)(ptr)))
+#endif
+#ifndef u64_to_ptr
+#define u64_to_ptr(x)	((void *)(uintptr_t)x)
+#endif
+
+#ifndef CLONE_ARGS_SIZE_VER0
```

Comment:
There also is no size attached to this. Is that on purpose?

Reply:
Looking at it now (one week later) it seems the whole file was started with other ideas than how I am using it right now.

The reason to call it `_clone_args` is to not collide with the kernel struct from the header (which is not included (which is wrong)). I always want to have a struct with the set_tid member to be able to test if it works. I am using the structure size defines (CLONE_ARGS_SIZE_VER1) mainly to check if the structure is defined at all.

Reply:
The way I define it in kernel tests is:
```c
#ifndef CLONE_ARGS_SIZE_VER0
#define CLONE_ARGS_SIZE_VER0 64
#endif

#ifndef __NR_clone3
#define __NR_clone3 -1
struct clone_args {
	__aligned_u64 flags;
	__aligned_u64 pidfd;
	__aligned_u64 child_tid;
	__aligned_u64 parent_tid;
	__aligned_u64 exit_signal;
	__aligned_u64 stack;
	__aligned_u64 stack_size;
	__aligned_u64 tls;
#define CLONE_ARGS_SIZE_VER1 80
	__aligned_u64 set_tid;
	__aligned_u64 set_tid_size;
#define CLONE_ARGS_SIZE_VER2 88
	__aligned_u64 cgroup;
};
#endif /* __NR_clone3 */
```
`CLONE_ARGS_SIZE_VER0` is only exported starting with kernel `v5.4`. So the reliable test for whether or not `clone3()` is available at all should be to look at `__NR_clone3`.